Technical/Performance-Benchmarking/performance-benchmarking-of-arangodb-single-instance-vs-split-databases-vs-clustering.md

```diff
@@ -0,0 +1,2030 @@
+
+****Clustering****
+
+For the Clustering configuration, our next step is a thorough review of our key usage within ArangoDB. The objective is to fully harness the capabilities of smart graphs, as detailed in ArangoDB’s guide on transforming graphs to smart graphs ([Transforming Graphs to SmartGraphs](https://arangodb.com/learn/graphs/transforming-graph-to-smartgraph/)). By optimizing key distribution and ensuring that related data is co-located on the same shard, we anticipate a significant boost in performance for graph-related queries. SmartGraphs offer a promising path to achieve horizontal scalability and improved query efficiency for our graph data models, especially in distributed database setups.
+
+**Example of the restructuring:**
+<pre language="json">
+{
+  _id:"edge/+49-1725310+49-11111111"
+  _key:"+49-1725310+49-11111111"
+}
+
+
+{_id:"1725310", cc:"+49"}
+{_id:"9725954444", cc:"+1"}
+
+{
+  _id:"edge/+49:17253109725954444:+1"
+  _key:"+49:17253109725954444:+1"
+}
+</pre>
+
+By adopting these two enhancements—splitting databases across individual ArangoDB instances and optimizing our Clustering configuration with smart graph technology we can achieve not only higher transactions per second (FTPS) but also to establish a more cost-effective infrastructure. These advancements will be pivotal in scaling our database capabilities to meet the expanding demands of our applications while simultaneously maximizing performance and managing costs.
+
+<span style="color:green; font-size: 24px;">****Final Completed Steps****</span>
+
+Continuing to refine and enhance the ArangoDB deployment strategies, the aim to implement significant improvements that will further optimize performance, scalability, and efficiency. The focus was on two main areas: SplitDbs and Clustering.
+
+****4 SplitDbs****
+
+The final test for the SplitDbs configuration involves splitting each database onto its own dedicated ArangoDB single instance. This approach showed distribute the read and write operations more evenly, leveraging multiple instances of ArangoDB. By isolating databases, we improved the overall performance by parallelizing the operations but also enhance the maintainability and scalability of the system. This split enabled the fine-tune resources and configurations specific to the workload and usage patterns of each database.
+
+[![](https://developer.sybrin.com/uploads/images/gallery/2024-04/scaled-1680-/GNKjWikQYwdHlLDP-image-1712573136010.png)](https://developer.sybrin.com/uploads/images/gallery/2024-04/GNKjWikQYwdHlLDP-image-1712573136010.png)
+
+<span style="color:green; font-size: 24px;">****JSON End-to-End Times****</span>
+
+Here, we will present the detailed JSON files that reflect the end-to-end processing times, providing a clear picture of how each transaction is handled by the system. This data is critical for understanding the latency and efficiency of each configuration.
```

Comment:
As with the other documents, I think these will benefit from a description of the attachments and an analysis of the results, or at least calling out some of the highlights.

Reply:
done